src/components/TodoList.tsx

```diff
@@ -0,0 +1,143 @@
+import { TodoItem } from './TodoItem';
+
+interface Props {
+  todos: Todo[];
+  deleteTodo: (id: number) => Promise<void>;
+  deleteDisabled: number[];
+  tempTodo: Todo | null;
+  deleteCompleted: () => Promise<void>;
+  clearDisabled: boolean;
+  handleChangeCompleted: (todo: Todo) => Promise<void>;
+  isCheckLoading: number;
+  showError: (message: string) => void;
+}
+
+enum Filter {
+  All = 'all',
+  Active = 'active',
+  Completed = 'completed',
+}
+
+export const TodoList: React.FC<Props> = ({
+  todos: todosData,
+  deleteTodo,
+  deleteDisabled,
+  tempTodo,
+  deleteCompleted,
+  clearDisabled,
+  handleChangeCompleted,
+  isCheckLoading,
+  showError,
+}) => {
+  const [todos, setTodos] = useState<Todo[]>(todosData);
+  const [filterBy, setFilterBy] = useState<Filter>(Filter.All);
+
+  useEffect(() => {
+    switch (filterBy) {
+      case Filter.Active:
+        setTodos(todosData.filter(todo => !todo.completed));
+        break;
+      case Filter.Completed:
+        setTodos(todosData.filter(todo => todo.completed));
+        break;
+      case Filter.All:
+      default:
+        setTodos(todosData);
+        break;
+    }
+  }, [filterBy, todosData]);
+
+  const filters = [
+    { label: 'All', value: Filter.All, href: '#/' },
+    { label: 'Active', value: Filter.Active, href: '#/active' },
+    { label: 'Completed', value: Filter.Completed, href: '#/completed' },
+  ];
+
+  return (
+    <>
+      <section className="todoapp__main" data-cy="TodoList">
+        {todos.map(todo => (
+          <TodoItem
+            key={todo.id}
+            showError={showError}
+            deleteTodo={deleteTodo}
+            deleteDisabled={deleteDisabled}
+            isCheckLoading={isCheckLoading}
+            handleChangeCompleted={handleChangeCompleted}
+            todo={todo}
+            setTodos={setTodos}
+          />
+        ))}
+        {tempTodo && (
+          <div key={tempTodo.id} data-cy="Todo" className="todo">
+            <label className="todo__status-label">
+              <input
+                data-cy="TodoStatus"
+                type="checkbox"
+                className="todo__status"
+                checked={tempTodo.completed}
+              />{' '}
+            </label>
+
+            <span data-cy="TodoTitle" className="todo__title">
+              {tempTodo.title}
+            </span>
+
+            <button
+              onClick={() => deleteTodo(tempTodo.id)}
+              type="button"
+              className="todo__remove"
+              data-cy="TodoDelete"
+              disabled={deleteDisabled.includes(tempTodo.id)}
+            >
+              ×
+            </button>
+
+            <div data-cy="TodoLoader" className="modal overlay is-active">
+              <div className="modal-background has-background-white-ter" />
+              <div className="loader" />
+            </div>
+          </div>
+        )}
+      </section>
+
+      <footer className="todoapp__footer" data-cy="Footer">
```

Comment:
What is `footer` doing in the `TodoList` component?
Create a separate component and use in an `App`. TodoList should get todos and render them, and know nothing about other UI parts